lib/chai/core/assertions.js

```diff
@@ -1532,6 +1532,56 @@ module.exports = function (chai, _) {
   });
 
   /**
+   * ### .oneOf(list)
+   *
+   * Assert that a value appears somewhere in the top level of array `list`.
+   *
+   *     expect('a').to.be.oneOf(['a', 'b', 'c']);
+   *     expect(9).to.not.be.oneOf(['z']);
+   *     expect([3]).to.not.be.oneOf([1, 2, [3]]);
+   *
+   *     var three = [3];
+   *     // for object-types, contents are not compared
+   *     expect(three).to.not.be.oneOf([1, 2, [3]]);
+   *     // comparing references works
+   *     expect(three).to.be.oneOf([1, 2, three]);
+   *
+   * @name oneOf
+   * @param {Array<*>} list
+   * @param {String} message _optional_
+   * @api public
+   */
+
+  function oneOf (list, msg) {
+    if (msg) flag(this, 'message', msg);
+    var expected = flag(this, 'object');
+    new Assertion(list).to.be.an('array');
+    var assertion = function () {
+        var matched = false;
+        // without a `deep` flag set, only examine top-level members. Do not recurse.
+        list.forEach(function (topLevelMember) {
+            // must be an exact reference to match. There are no comparisons for content!
+            if (topLevelMember === expected) {
+                matched = true;
+                return false;
+            }
+        });
+        return matched;
+    };
```

Comment:
@keithamus I put it back...I was halfway done with `_.isEquals`-style object comparison (content matches, not references). I forgot that I could use the simpler `indexOf` once you asked for reference-based matching.

I force pushed it up to keep the history tidy.